### PR TITLE
parallel tests with xdist

### PR DESCRIPTION
**Pull request by burnettk**
`./bin/tests-par` runs about four times faster than `poet test` for me.

### CODE REVIEW

File: spiffworkflow-backend/bin/tests-par [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+set -o errtrace -o errexit -o nounset -o pipefail
+
+if [[ ! -f ./src/instance/db_unit_testing_gw0.sqlite3 ]] ; then
+  >&2 echo -e "ERROR: please run the following command first in order to set up and migrate the sqlite unit_testing database:\n\n\tSPIFFWORKFLOW_BACKEND_DATABASE_TYPE=sqlite ./bin/recreate_db clean"
+  exit 1
+fi
+
+# check if python package hey is installed
+if ! python -c "import xdist" &>/dev/null; then
+  >&2 echo -e "ERROR: please install the python package pytest-xdist first:\n\n\tpip install pytest-xdist"
```
Comment (jbirddog) [inline spiffworkflow-backend/bin/tests-par:17]: Could this be a dev dependency via poetry?

**Comment by burnettk** [comment]
yeah, true, and that shouldn't affect prod images.

File: spiffworkflow-backend/bin/tests-par [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+function error_handler() {
+  >&2 echo "Exited with BAD EXIT CODE '${2}' in ${0} script at line: ${1}."
+  exit "$2"
+}
+trap 'error_handler ${LINENO} $?' ERR
+set -o errtrace -o errexit -o nounset -o pipefail
+
+if [[ ! -f ./src/instance/db_unit_testing_gw0.sqlite3 ]] ; then
+  >&2 echo -e "ERROR: please run the following command first in order to set up and migrate the sqlite unit_testing database:\n\n\tSPIFFWORKFLOW_BACKEND_DATABASE_TYPE=sqlite ./bin/recreate_db clean"
```
Comment (jbirddog) [inline spiffworkflow-backend/bin/tests-par:11]: Also any reason not to just run this command in this scenario? btw very pumped about this landing.

**Review by jbirddog — APPROVED** [review]

File: spiffworkflow-backend/bin/tests-par (hunk 2 above)
Comment (burnettk) [reply spiffworkflow-backend/bin/tests-par:11]: i think it'll clear the development database, too, so might be a bit dangerous. could prompt, i guess.

Comment (jbirddog) [reply spiffworkflow-backend/bin/tests-par:11]: Gotcha - if maybe dangerous then sure either way is fine. Less code not to prompt I suppose.

**Pull request merged**